x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/RBACEngineTests.java

```diff
@@ -143,33 +142,35 @@ public void testSameUserPermission() {
     }
 
     public void testSameUserPermissionDoesNotAllowNonMatchingUsername() {
-        final User authUser = new User("admin", new String[] { "bar" });
-        final User user = new User("joe", null, authUser);
+        final User authUser = new User("admin", "bar");
+        final User user = new User("joe");
         final boolean changePasswordRequest = randomBoolean();
         final String username = randomFrom("", "joe" + randomAlphaOfLengthBetween(1, 5), randomAlphaOfLengthBetween(3, 10));
         final TransportRequest request = changePasswordRequest
             ? new ChangePasswordRequestBuilder(mock(Client.class)).username(username).request()
             : new AuthenticateRequestBuilder(mock(Client.class)).username(username).request();
         final String action = changePasswordRequest ? ChangePasswordAction.NAME : AuthenticateAction.NAME;
-        final Authentication authentication = mock(Authentication.class);
-        final Authentication.RealmRef authenticatedBy = mock(Authentication.RealmRef.class);
-        when(authentication.getUser()).thenReturn(user);
-        when(authentication.getAuthenticatedBy()).thenReturn(authenticatedBy);
+
         final String authenticationType = changePasswordRequest
             ? randomFrom(ReservedRealm.TYPE, NativeRealmSettings.TYPE)
             : randomAlphaOfLengthBetween(4, 12);
```

Comment:
I don't think we need this conditional do we?
This is used for the `authUser` - we don't care what realm type it has, we only care which realm `user` was `lookedUpBy`.

```suggestion
        final String authenticationType = randomAlphaOfLengthBetween(4, 12);
```

Though `authenticationRealmType` is probably a better name.

Reply:
You are right. It does not matter. I inlined it with simply `randomAlphaOfLengthBetween(4, 12)`.